### PR TITLE
Update docs for NetworkPolicy port range to GA for 1.25

### PR DESCRIPTION
**Pull request by reylejano**
This PR updates docs for the NetworkPolicy port range enhancement that is graduating to stable in 1.25 by the following:
- updates the feature state on the [Targeting a Range of Ports section on the Network Policies page](https://kubernetes.io/docs/concepts/services-networking/network-policies/#targeting-a-range-of-ports) to 1.25
- updates the Feature Gates page for _NetworkPolicyEndPort_ and moves entries from _Feature gates for Alpha or Beta features_ section to the _Feature gates for graduated or deprecated features_ section and adding an entry for 1.25 to GA 

Enhancement issue: https://github.com/kubernetes/enhancements/issues/2079
KEP: https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/2079-network-policy-port-range

/sig network
/milestone 1.25
/assign @kcmartin @rikatz

### CODE REVIEW

**Comment by reylejano** [comment]
@rikatz and @kubernetes/sig-network-pr-reviews  , the NetworkPolicy port range enhancement is tracked to be graduated in 1.25
Please review if the changes in this PR to k/website are correct for the NetworkPolicy port range going to stable.
Are there any additional docs that need to be updated for NetworkPolicy port range?

**Comment by cici37** [comment]
It might need a little bit rewording inside the same section with feature going GA:
https://github.com/kubernetes/website/blob/eb860d5d5cbc7f4e98f03aca7362e85eac55d26c/content/en/docs/concepts/services-networking/network-policies.md?plain=1#L231

**Comment by reylejano** [comment]
> It might need a little bit rewording inside the same section with feature going GA:
> 
> https://github.com/kubernetes/website/blob/eb860d5d5cbc7f4e98f03aca7362e85eac55d26c/content/en/docs/concepts/services-networking/network-policies.md?plain=1#L231

Nice catch @cici37 
@rikatz is deleting the instructions to disable the feature gate correct

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -395,6 +393,9 @@ different Kubernetes components.
 | `MountPropagation` | `true` | GA | 1.12 | - |
 | `NamespaceDefaultLabelName` | `true` | Beta | 1.21 | 1.21 |
 | `NamespaceDefaultLabelName` | `true` | GA | 1.22 | - |
+| `NetworkPolicyEndPort` | `false` | Alpha | 1.21 | 1.21 |
+| `NetworkPolicyEndPort` | `true` | Beta | 1.22 |  |
```
Comment (rikatz) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:397]: Don't we wanna add the "until" here for v1.24?

**Comment by rikatz** [comment]
@reylejano thanks and sorry again for the delay.
This lgtm, just left a minor small comment. I was worried that we cover the fact that this feature just works if the NPP/CNI supports it, but this callout is already on the page
/lgtm
/approve
/hold
@thockin I guess this one can go, as we discussed in some of sig-net meetings. The note on endPort being CNI dependant is there.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md (hunk 1 above)
Comment (sftim) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:397]: Yep, we should add that.

Comment (rikatz) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:397]:
```suggestion
| `NetworkPolicyEndPort` | `true` | Beta | 1.22 | 1.24 |
```

**Review by rikatz — COMMENTED** [review]
Making life easier and adding the change directly :) feel free to commit and merge it

**Comment by reylejano** [comment]
Unholding to let this PR merge per this [comment](https://github.com/kubernetes/website/pull/35826#pullrequestreview-1072120326)
/hold cancel

**Pull request merged**